src/utils/disks/prepareDisks.ts

```diff
@@ -1,70 +1,155 @@
-import type {TPDiskInfo, TPDiskStateInfo} from '../../types/api/pdisk';
-import type {TVDiskStateInfo} from '../../types/api/vdisk';
+import type {TPDiskStateInfo} from '../../types/api/pdisk';
+import type {TVDiskStateInfo, TVSlotId} from '../../types/api/vdisk';
 import {stringifyVdiskId} from '../dataFormatters/dataFormatters';
 
 import {calculatePDiskSeverity} from './calculatePDiskSeverity';
 import {calculateVDiskSeverity} from './calculateVDiskSeverity';
 import {getPDiskType} from './getPDiskType';
+import {getPDiskId, isFullVDiskData} from './helpers';
 import type {PreparedPDisk, PreparedVDisk} from './types';
 
-export function prepareVDiskData(vdiskState: TVDiskStateInfo = {}): PreparedVDisk {
+export function prepareWhiteboardVDiskData(
+    vDiskState: TVDiskStateInfo | TVSlotId = {},
+): PreparedVDisk {
+    if (!isFullVDiskData(vDiskState)) {
+        const {NodeId, PDiskId, VSlotId} = vDiskState;
+
+        const StringifiedId = stringifyVdiskId({
+            NodeId,
+            PDiskId,
+            VSlotId,
+        });
+
+        return {
+            StringifiedId,
+            NodeId,
+            PDiskId,
+            // Replace VSlotId with VDiskSlotId to match with PreparedVDisk type
+            VDiskSlotId: VSlotId,
+        };
+    }
+
+    const {
+        PDisk,
+        PDiskId,
+        VDiskId,
+        NodeId,
+        Donors,
+        AvailableSize,
+        AllocatedSize,
+        ...restVDiskFields
+    } = vDiskState;
+
     // Prepare PDisk only if it is present
-    const PDisk = vdiskState.PDisk
-        ? preparePDiskData({
-              ...vdiskState.PDisk,
-              NodeId: vdiskState.PDisk.NodeId ?? vdiskState.NodeId,
-          })
+    const preparedPDisk = PDisk
+        ? prepareWhiteboardPDiskData({...PDisk, NodeId: PDisk?.NodeId ?? NodeId})
         : undefined;
 
-    const PDiskId = vdiskState.PDiskId ?? PDisk?.PDiskId;
+    const actualPDiskId = PDiskId ?? PDisk?.PDiskId;
```

Comment:
I would expect that preparedPDisk contains actial Id

Reply:
There might be no PDisk inside VDisk in some cases. For example, for `viewer/nodes` endpoint `VDisks` and `PDisks` are located in separate fields